jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -133,8 +133,18 @@ pointer_typet select_pointer_typet::specialize_generics(
 }
 
 /// Return the first concrete type instantiation if any such exists. This method
-/// is only to be called when the `specialize_generics` cannot find an
-/// instantiation due to a loop in its recursion.
+/// is only to be called when \ref select_pointer_typet::specialize_generics
+/// cannot find an instantiation due to a loop in its recursion.
+///
+/// Example:
+/// `class MyGeneric<T,U> { MyGeneric<U,T> gen, T t}`
+/// When instantiating `MyGeneric<Integer,String> my` we need to for example
+/// resolve the type of `my.gen.t`. The map would in this context contain
+/// - T -> (Integer, U)
+/// - U -> (String, T)
+///
+/// Note that the top of the stacks for T and U create a recursion T -> U,
+/// U-> T. We want to break it and specialize `T my.gen.t` to `String my.gen.t`.
```

Comment:
:+1: Thanks for adding this example! I didn't realise that the generic specialisation map doesn't actually keep track of the context in which the parameters are specialised to certain types. Can this limitation cause any problems?

Reply:
From offline discussion: there is no real limitation, for each parameter name of the form `<classname>::<parametername>` we keep a stack of types that determine what the parameter is specialized with the given context depth. When the top of the stacks of parameters form a cycle (`T->U,U->T`) that's when we need to dig a little deeper to actually find the non-parametric types (if they are available).